components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserSelfSignUpHandler.java

```diff
@@ -0,0 +1,232 @@
+                        claims.add(claim);
+
+                        identityStore.updateUserClaims(user.getUniqueUserId(), claims, null);
+                    } catch (IdentityStoreException e) {
+                        throw Utils.handleServerException(ErrorCodes.FAILED_ACCOUNT_LOCK, user
+                                .getUniqueUserId(), e);
+                    }
+
+                    UserRecoveryDataStore userRecoveryDataStore = JDBCRecoveryDataStore.getInstance();
+                    userRecoveryDataStore.invalidateByUserUniqueId(user.getUniqueUserId());
+
+                    String secretKey = Utils.generateUUID();
+                    UserRecoveryData recoveryDataDO = new UserRecoveryData(user.getUniqueUserId(), secretKey,
+                                                                           RecoveryScenarios.SELF_SIGN_UP,
+                                                                           RecoverySteps.CONFIRM_SIGN_UP);
+                    userRecoveryDataStore.store(recoveryDataDO);
+
+                    if (config.isNotificationInternallyManaged()) {
+
+                        Property[] arbitraryProperties = (Property[]) eventProperties.get(SELF_SIGN_UP_PROPERTIES);
+                        Utils.triggerNotification(eventService, user.getUniqueUserId(),
+                                                  NOTIFICATION_TYPE_ACCOUNT_CONFIRM,
+                                                  secretKey,
+                                                arbitraryProperties);
+                    } else {
+                        eventContext.addParameter(IdentityStoreConstants.UNIQUE_USED_ID, user.getUniqueUserId());
+                        eventContext.addParameter(CONFIRMATION_CODE, secretKey);
+                    }
+                } else {
+
+                    if (log.isDebugEnabled()) {
+                        log.debug("Changing user lifecycle state to {} for user: {}", UserState
+                                .UNLOCKED__UNVERIFIED.toString(), user.getUniqueUserId());
+                    }
+                    LifecycleOperationManager.executeLifecycleEvent(UserState.UNLOCKED__UNVERIFIED.toString(),
+                                                                    domainUID, domainUID, null);
+                    identityStore.setUserState(user.getUniqueUserId(), UserState.UNLOCKED__UNVERIFIED.toString());
+                }
+            } catch (LifecycleException e) {
+                throw Utils.handleServerException(ErrorCodes.FAILED_LIFECYCLE_EVENT, null, e);
+            } catch (UserNotFoundException e) {
+                throw Utils.handleServerException(ErrorCodes.USER_NOT_FOUND, null, e);
+            } catch (IdentityStoreException e) {
+                throw Utils.handleServerException(ErrorCodes.FAILED_USER_STATE_UPDATE, null, e);
+            }
+        }
+
+        if (IdentityStoreInterceptorConstants.POST_AUTHENTICATE.equals(event.getEventName())) {
```

Comment:
we handle two different events in this method(line 91 and 184). It is better to handle them 
1. in private specific methods, or 
2. Separate Hander implementations